packages/lumx-react/src/components/text-field/TextField.tsx

```diff
@@ -71,13 +71,13 @@ interface TextFieldProps extends GenericProps {
     inputRef?: RefObject<HTMLInputElement> | RefObject<HTMLTextAreaElement>;
 
     /** Text field value. */
-    value: string;
+    value: string | number;
 
     /** A ref that will be passed to the wrapper element. */
     textFieldRef?: RefObject<HTMLDivElement>;
 
     /** Text field value change handler. */
-    onChange(value: string): void;
+    onChange(value: any): void;
```

Comment:
```suggestion
    onChange(value: string | number): void;
```
??
The problem is that the native input value is always string when accessed so I guess we need to do a `parseFloat` of the native value if the input type is `number`

Reply:
How do you handle the following?
```js
const [foo, setFoo] = useState('foo');

// ...

    onChange={setFoo}

// ...
```

The signature are not the same.

I find the following quite ugly:
`onChange={v => setFoo(v as string)}`

Reply:
It's possible if we change the TextFieldProps to do something like:
```tsx
type TextFieldProps =
  | { type: 'number'; value?: number; onChange(value: number): void; }
  | { value?: string; onChange(value: string): void; }
```
^^
It could be the occasion to also do something like:

```tsx
type TextFieldProps =
  | ( { multiline: true } & TextareaHTMLAttributes<HTMLTextAreaElement>)
  | ( { multiline?: false } & InputHTMLAttributes<HTMLInputElement>)
```
So that the native props depends on the actual native input used

Reply:
It's possible if we change the TextFieldProps to do something like:
```tsx
type TextFieldProps =
  | { type: 'number'; value?: number; onChange(value: number): void; }
  | { value?: string; onChange(value: string): void; }
```
^^
It could be the occasion to also do something like:

```tsx
type TextFieldProps =
  | ( { multiline: true } & TextareaHTMLAttributes<HTMLTextAreaElement>)
  | ( { multiline?: false } & InputHTMLAttributes<HTMLInputElement>)
```
So that the forwarded props depends on the actual native input used

Reply:
I didn't succeed in this kind of implementation. 
I had errors with the onChange function.
Can we postpone these changes in another PR in order to give us some time to see exactly how to do these improvements?

Reply:
Ok but actually, now that I'm thinking about it the `onChange` still returns a value as a `string` rven if you give a `value` as a `number` because the native input value is always a `string`. 
So either we do a `parseFloat` when `type==='number'` (and keep `onChange(value: any): void`) or we keep the type `onChange(value: string): void;` 
What do you think @radiuso ?

Reply:
Yes, good point.
I think we can keep the `string` type. Maybe we want to use an integer or a float, the choice should not be done by the component.

Reply:
Ok
But parseFloat works with integer as well as float number ^^